GATK_QuickByte.md

```diff
@@ -262,46 +256,93 @@ This first step is optional, but here we separate out indels and SNPs. Note that
+	
+	# make our interval list
+	cut -f 1 ${reference}.fa.fai > $src/intervals.list
+	
+	while read sample; do
+		mkdir ${src}/gvcfs/${sample}
+		cat $src/intervals.list | env_parallel --sshloginfile $PBS_NODEFILE \
+			'gatk --java-options "-Xmx6g" HaplotypeCaller \
+			-R ${reference}.fa \
+			-I $src/bams/${sample}_recal.bam \
+			-O $src/gvcfs/${sample}/${sample}_{}_raw.g.vcf.gz \
+			-L {} \
+			-ERC GVCF'
+	done < $src/sample_list
+	
+You'll run then run CombineGVCFs. For each interval, you'll make a list of GVCF file paths for each sample you're including (the while loop below).
+
+	cat $src/intervals.list | env_parallel --sshloginfile $PBS_NODEFILE \
+		'interval_list=""
+		# loop to generate list of sample-specific intervals to combine
+		while read sample; do
+			interval_list="${interval_list}-V ${src}/gvcfs/${sample}/${sample}_{}_raw.g.vcf.gz "
+		done < $src/sample_list
+		gatk --java-options "-Xmx6g" CombineGVCFs \
+			-R ${reference}.fa \
+			${interval_list} \
+			-O $src/gvcfs/combined_intervals/{}_raw.g.vcf.gz'
+Next, you run GenotypeGVCFs to get VCFs to gather afterwards. No fancy lists needed!
+
+	cat $src/intervals.list | env_parallel --sshloginfile $PBS_NODEFILE \
+		'gatk --java-options "-Xmx6g" GenotypeGVCFs \
+			-R ${reference}.fa \
+			-V $src/gvcfs/combined_intervals/{}_raw.g.vcf.gz \
+			-O $src/combined_vcfs/intervals/{}_genotyped.vcf.gz'
+			
+The final (gather) step uses GatherVcfs, for which we'll make a file containing the paths to all input genotyped VCFs (generated in the while loop).
+
+	> $src/combined_vcfs/gather_list
+	while read interval; do
+   		echo ${src}/combined_vcfs/intervals/${interval}_genotyped.vcf.gz >> \
+       			$src/combined_vcfs/gather_list
+	done < $src/chromosomes.list
+	
+	gatk GatherVcfs \
+    		-I $src/combined_vcfs/gather_list \
+    		-O combined_vcfs/combined_vcf.vcf.gz
+
 ## Sample PBS Script ##
 
 Here is a sample PBS script combining everything we have above, with as much parallelization as possible. One reason to break up steps like we did is for improved checkpointing (without having to write code checking if files are already present). Once you are finished running a block of code, you can just comment it out. Similarly, if you can only get part way through your sample list, you can copy it and remove samples that have already completed a given step.
 
 	module load miniconda3-4.7.12.1-gcc-4.8.5-lmtvtik
 	source activate gatk-env
-
-	module load parallel-20170322-gcc-4.8.5-2ycpx7e
-	source $(which env_parallel.bash)
-
+	
 	src=$PBS_O_WORKDIR
-	reference=${src}/reference
+	reference=${src}/reference.fa
 	
 	# Trimming section
 	adapters=~/.conda/pkgs/trimmomatic-0.39-1/share/trimmomatic-0.39-1/adapters/TruSeq3-PE.fa
```

Comment:
adapters=$src/TruSeq3-PE.fa

Also a note, the pbs script uses echo for trimmomatic and I believe the extra spaces afterwards causes it to fail. However, the code below does work.